### ISSUE TITLE
Get fully qualified symbol at point

### ISSUE THREAD

**Issue by agzam** [snoe/clojure-lsp#192]
I often need to find out the fully qualified name of a symbol at point (In Emacs), so I do this:

```emacs-list
(when (lsp--capability :hoverProvider)
  (let ((s (-some->> (lsp--text-document-position-params)
             (lsp--make-request "textDocument/hover")
             (lsp--send-request)
             (gethash "contents")
             (gethash "value"))))
    (string-match "\\(```.*\n\\)\\(.*\\)\n\\(```\\)" s)
    (string-trim (match-string 2 s))))
```
But this does not always work for some reason, and honestly, feels like a cheap hack. Can someone suggest a better way, please?

**Comment by ericdallo** [comment]
@agzam I think you should ask that on [lsp-mode](https://github.com/emacs-lsp/lsp-mode) or open an issue there :)

**Comment by agzam** [comment]
Yeah, I guess I'm gonna have to move it there. I just thought maybe someone here has dealt with something like that before.

**Comment by agzam** [comment]
After talking to lsp-mode folks, I think I'm gonna have to re-open it. This is what @yyoncho says:

> 2 options:
> Your ad-hoc solution
> File a bug in clojure-lsp to expose a custom method. The protocol itself does not provide that info in structured format
> By structured I mean machine readable. The protocol provides only human readable one via onHover

**Comment by ericdallo** [comment]
@agzam I see, do you have an example of how would be the expected experience for that feature?

**Comment by agzam** [comment]
I'm not sure what you mean. Just a custom server method to extract a fully-qualified symbol at point. The clients then can implement a call to retrieve that info. This could be useful for example when you need to copy&paste fully qualified name (useful in chat's for example), for hints in the mode-line/status-line (e.g.: via eldoc in Emacs).

Then, the clients can also extend it for example to automatically `require` the namespace for a given symbol at point. Very often when I see a symbol `foo` being used in namespace `A`, and need to use the same symbol in namespace `B`, there's no easy way to automatically require that symbol, the workflow is always somewhat messy. Would be nice if one could just "copy&paste" a "pointer" to the symbol, and the IDE (using lsp) would figure out that it is not just some `foo` but exactly `my-service.core/foo` and would add a `require` statement for it.

Unfortunately, it seems lsp protocol does not offer any useful instrument for seemingly basic functionality like that. We can of course submit an issue, but it will take long time for them to implement it. Perhaps it's the best if a custom method was added to clojure-lsp with the possibility of extending it in clients. And maybe someday, when lsp-protocol gets a native implementation for that, we could adjust it accordingly.

**Comment by ericdallo** [comment]
Yeah, I see no problem in creating a custom method for that (I could open a PR on lsp-mode after clojure-lsp), but I'm not sure if @snoe agree with adding this since this is not part of LSP spec :/
